mtags/src/main/scala-2/scala/meta/internal/pc/Keywords.scala

```diff
@@ -19,6 +20,12 @@ trait Keywords { this: MetalsGlobal =>
 
     lazy val notInComment = checkIfNotInComment(pos, text, latestEnclosing)
 
+    lazy val reverseTokens: Iterator[Token] =
+      text.substring(0, pos.start).tokenize.toOption match {
```

Comment:
I understand, but it would be a problem (Metals won't complete `extends` keyword), for example in the case of

```scala
class Foo(...)
  exte@@
```

but it's a trade-off, and I don't think people write `extends` in a newline when they're trying to define a class.

Maybe we can go with `text.substring(linestart, pos.start)` and we can change the behavior later if someone complains about it.

Reply:
done https://github.com/scalameta/metals/pull/4416/commits/1e66b779a1e875d8c3308209eaa2443f75950744
but maybe we should share the tokenized result with the `notInComment`